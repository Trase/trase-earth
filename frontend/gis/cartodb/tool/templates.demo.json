[
  {
    "version": "0.0.1",
    "name": "ar_provinces_20191118",
    "layergroup": {
      "version": "1.0.1",
      "layers": [
        {
          "type": "cartodb",
          "options": {
            "sql": "SELECT * FROM \"p2cs-sei\".ar_provinces_20191118",
            "cartocss": "#layer { polygon-opacity: 0; line-color: #000000; line-width: 1; line-opacity: 1; } #layer[zoom > 3]::labels { text-name: [prov_name]; text-face-name: 'Open Sans Regular'; text-size: 10; text-fill: #FFFFFF; text-halo-fill: black; text-halo-radius: 1; text-character-spacing: 3; text-placement: point; text-allow-overlap: false; text-wrap-width: 45; }",
            "cartocss_version": "2.3.0"
          }
        }
      ]
    }
  },
  {
    "version": "0.0.1",
    "name": "ar_biomes_20191113",
    "layergroup": {
      "version": "1.0.1",
      "layers": [
        {
          "type": "cartodb",
          "options": {
            "sql": "SELECT * FROM \"p2cs-sei\".ar_biomes_20191113",
            "cartocss": "#layer { polygon-opacity: 0; line-color: #000000; line-width: 0.5; line-opacity: 1; } #layer[zoom > 4]::labels { text-name: [biome_name]; text-face-name: 'Open Sans Regular'; text-size: 15; text-fill: #FFFFFF; text-allow-overlap: false; text-placement: interior; text-halo-fill: black; text-halo-radius: 0.7; text-character-spacing: 3; } ",
            "cartocss_version": "2.3.0"
          }
        }
      ]
    }
  },{
  "version": "0.0.1",
  "name": "argentina_protected_areas_20191117",
  "layergroup": {
    "version": "1.0.1",
    "layers": [
      {
        "type": "cartodb",
        "options": {
          "sql": "SELECT * FROM \"p2cs-sei\".argentina_protected_areas_20191117",
          "cartocss": "#layer { polygon-opacity: 0.5; polygon-fill: ramp([type], (#8dd3c7, #ffffb3, #bebada, #fb8072, #80b1d3, #fdb462, #b3de69), (\"National Protected Areas\", \"National Defense Reservers\", \"Ramsar\", \"Natural World Heritage Sites\", \"Provincial Protected Areas\", \"Biosphere Reserves\"), \"=\"); line-color: ramp([type], (#8dd3c7, #ffffb3, #bebada, #fb8072, #80b1d3, #fdb462, #b3de69), (\"National Protected Areas\", \"National Defense Reservers\", \"Ramsar\", \"Natural World Heritage Sites\", \"Provincial Protected Areas\", \"Biosphere Reserves\"), \"=\"); line-width: 1; line-opacity: 1; }",
          "cartocss_version": "2.3.0"
        }
      }
    ]
  }
}
]
