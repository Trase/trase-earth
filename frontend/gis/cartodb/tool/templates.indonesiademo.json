[
  {
    "version": "0.0.1",
    "name": "id_provinces",
    "layergroup": {
      "version": "1.0.1",
      "layers": [
        {
          "type": "cartodb",
          "options": {
            "sql": "SELECT * FROM indonesia_provinces_boundaries_20190614",
            "cartocss": "#layer { polygon-opacity: 0; line-color: #000000; line-width: 1; line-opacity: 1; ::outline { line-color: #FFFFFF; line-width: 1; line-opacity: 0.5; } [zoom>4]::labels { text-face-name: 'Open Sans Regular'; text-name: [prov]; text-size: 12; text-allow-overlap: true; text-placement: point; text-fill: #000000; text-character-spacing: 0; [prov='SULAWESI UTARA'] { text-dy: -10; } [prov='BENGKULU'] { text-dy: 10; } [prov='SULAWESI TENGGARA'] { text-dy: 10; } [prov='BANTEN'] { text-dx: -10; } [prov='DKI JAKARTA'] { text-dy: -10; } [prov='JAWA BARAT'] { text-dy: -10; } [prov='JAWA TIMUR'] { text-dx: 10; text-dy: -10; } [prov='NUSA TENGGARA BARAT'] { text-dy: 10; } [prov='NUSA TENGGARA TIMUR'] { text-dx: 5; } } } ",
            "cartocss_version": "2.3.0"
          }
        }
      ]
    }
  }
]
