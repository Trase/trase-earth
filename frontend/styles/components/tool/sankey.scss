@import 'styles/settings';

.c-sankey {
  .sankey-node {
    cursor: pointer;

    .sankey-node-rect {
      fill: $white-two;
      stroke: $pinkish-grey-two;
      stroke-width: 1px;
    }

    .sankey-node-label {
      font-family: $font-family-1;
      font-size: $font-size-regular;
      text-transform: capitalize;
      text-anchor: middle;
      fill: $charcoal-grey;
      pointer-events: none;
    }

    &.-is-aggregated > .sankey-node-rect {
      // the horizontal lines pattern is defined in the <defs> section of the SVG in flows.ejs
      cursor: zoom-in;
      fill: url('#isAggregatedPattern');
    }

    &.-is-domestic > .sankey-node-rect {
      fill: $gray;
    }

    &.-is-alone-in-column {
      pointer-events: none;

      & > .sankey-node-rect {
        fill: $background-white;
      }
    }

    &.-selected {
      // disabled for now because of a Firefox rendering bug
      //filter: url(#drop-shadow); //ignore sasslint warning here

      .sankey-node-rect {
        fill: $white;
        stroke: $charcoal-grey;
      }
    }

    &.-highlighted {

      .sankey-node-rect {
        fill: $manilla;
      }
    }
  }


  .sankey-links .sankey-link {
    opacity: .1;
    fill: none;
    stroke: $charcoal-grey;

    &.-hover {
      opacity: .7;
    }

    @for $i from 1 through length($recolorby-colors) {
      &.#{nth($recolorby-colors-names, $i)} {
        opacity: .5;
        stroke: #{nth($recolorby-colors, $i)};
      }

      &.#{nth($recolorby-colors-names, $i)}.-hover {
        opacity: 1;
      }
    }

    @for $i from 1 through length($recolorgroup-colors) {
      &.#{nth($recolorgroup-colors-names, $i)} {
        opacity: .5;
        stroke: #{nth($recolorgroup-colors, $i)};
      }

      &.#{nth($recolorgroup-colors-names, $i)}.-hover {
        opacity: 1;
      }
    }
  }
}
