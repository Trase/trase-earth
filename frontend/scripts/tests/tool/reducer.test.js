import { SET_NODE_ATTRIBUTES } from 'react-components/tool/tool.actions';
import reducer from 'react-components/tool-links/tool-links.reducer';
import initialState from 'react-components/tool-links/tool-links.initial-state';

describe(SET_NODE_ATTRIBUTES, () => {
  it('adds node attributes', () => {
    const attributes = [
      { node_id: 1, attribute_id: 'SOY_DEFORESTATION', attribute_type: 'quant' },
      { node_id: 1, attribute_id: 'SMALLHOLDER_DOMINANCE', attribute_type: 'ind' },
      { node_id: 2, attribute_id: 'SMALLHOLDER_DOMINANCE', attribute_type: 'ind' },
      { node_id: 3, attribute_id: 'SMALLHOLDER_DOMINANCE', attribute_type: 'ind' }
    ];
    const action = {
      type: SET_NODE_ATTRIBUTES,
      payload: { data: attributes }
    };
    const newState = reducer(initialState, action);
    expect(newState).toMatchSnapshot();
  });

  it('removes node attributes', () => {
    const state = {
      ...initialState,
      data: {
        ...initialState.data,
        nodeAttributes: {}
      }
    };
    const action = {
      type: SET_NODE_ATTRIBUTES,
      payload: { data: [] }
    };
    const newState = reducer(state, action);
    expect(newState).toMatchSnapshot();
  });
});
