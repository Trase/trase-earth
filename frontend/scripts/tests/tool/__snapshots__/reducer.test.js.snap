// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SET_NODE_ATTRIBUTES adds node attributes 1`] = `
Object {
  "currentQuant": null,
  "data": Object {
    "columns": null,
    "links": null,
    "nodeAttributes": Object {
      "1": Object {
        "indSMALLHOLDER_DOMINANCE": Object {
          "attribute_id": "SMALLHOLDER_DOMINANCE",
          "attribute_type": "ind",
          "node_id": 1,
        },
        "quantSOY_DEFORESTATION": Object {
          "attribute_id": "SOY_DEFORESTATION",
          "attribute_type": "quant",
          "node_id": 1,
        },
      },
      "2": Object {
        "indSMALLHOLDER_DOMINANCE": Object {
          "attribute_id": "SMALLHOLDER_DOMINANCE",
          "attribute_type": "ind",
          "node_id": 2,
        },
      },
      "3": Object {
        "indSMALLHOLDER_DOMINANCE": Object {
          "attribute_id": "SMALLHOLDER_DOMINANCE",
          "attribute_type": "ind",
          "node_id": 3,
        },
      },
    },
    "nodeHeights": null,
    "nodes": null,
    "nodesByColumnGeoId": null,
  },
  "detailedView": false,
  "expandedNodesIds": Array [],
  "flowsLoading": false,
  "forcedOverview": false,
  "highlightedNodeId": null,
  "isSearchOpen": false,
  "noLinksFound": false,
  "selectedBiomeFilterName": null,
  "selectedColumnsIds": null,
  "selectedNodesIds": Array [],
  "selectedRecolorByName": null,
  "selectedResizeByName": null,
}
`;

exports[`SET_NODE_ATTRIBUTES removes node attributes 1`] = `
Object {
  "currentQuant": null,
  "data": Object {
    "columns": null,
    "links": null,
    "nodeAttributes": null,
    "nodeHeights": null,
    "nodes": null,
    "nodesByColumnGeoId": null,
  },
  "detailedView": false,
  "expandedNodesIds": Array [],
  "flowsLoading": false,
  "forcedOverview": false,
  "highlightedNodeId": null,
  "isSearchOpen": false,
  "noLinksFound": false,
  "selectedBiomeFilterName": null,
  "selectedColumnsIds": null,
  "selectedNodesIds": Array [],
  "selectedRecolorByName": null,
  "selectedResizeByName": null,
}
`;
